pkg/template/template.go

```diff
@@ -25,16 +25,18 @@ const (
 // Template is the representation of a template.
 type Template struct {
 	colorEnabled bool
+	isTTY        bool
 	output       io.Writer
 	tmpl         *template.Template
 	tp           tableprinter.TablePrinter
 	width        int
 }
 
 // New initializes a Template.
-func New(w io.Writer, width int, colorEnabled bool) Template {
+func New(w io.Writer, width int, colorEnabled, isTTY bool) Template {
```

Comment:
Fair, but then aren't we pushing the work to everyone who uses a template? Alternatively, what if we add an `isTTY` (or something) function that template conditions can use? Or is the assumption that the caller is always in control of both the template and the output, so they can easily and always adapt as needed?

Reply:
I think that last time we've had this discussion was in https://github.com/cli/cli/pull/3519#issuecomment-900592822

We've opted to always default to TTY output for consistency so that the user of `--template` doesn't have to guess the actual format of the final output depending on the end-user's execution context. If the gh developer wants separate formats for TTY vs script output, yes in theory they should use two separate templates.